rfc/parameterizedMetrics.md

```diff
@@ -0,0 +1,124 @@
+
+With parameterized metrics, open-ended variations may be supported for a given conceptual metric calculation, simplifying configuration and possibly unlocking otherwise impractical freedom in metric calculation.
+
+
+The structure of a parameterized metric:
+----------------------------------------
+
+```
+metricName(parameterName=value)
+```
+
+metricName -> an api metric name which should appear as a resource at the `/v1/metrics/metricName` endpoint.
+
+Balanced outer parentheses are the signal that a metric is to be processed as parameterized metric.  Imbalanced parentheses or unescaped internal parentheses will probably result in request errors.
+
+parameterName -> an identifier for a parameter supported by this metric.  Identifier names should follow the same format rules as api resource identifiers.
+
+value -> a value to be used for creating a modified version of baseMetric.  URL encoding may be required if some of the characters in the value contain special characters.  Escaping values containing ampersand is mandatory and it may also be necessary to encode values containing comma or parentheses depending on implementation.
+
+If multiple parameter names are used for this dimension they should be joined by commas with no intermediate spaces (similar to the filter value syntax)
+
+i.e.
+
+```
+metricName(parameterName=value,otherParameter=otherValue)
+```
+
+Response schema
+---------------
+
+Unless aliasing is supported and used (see below), responses will use the full string of the metric expression as the column name for values resulting from parametrized metric evaluation.
+
+
+Reserved parameter names
+------------------------
+
+Two optional features which may be supported in some fili implementations are 'filter' and 'AS'.  These parameter names should be treated as reserved when configuring metrics.
+
+Dynamic aggregation filters, if supported, use the parameter name `filter`. Values will be parsed according to the same rules as the entries in the `filters` query parameter.  The expected behavior of the filter parameter is that rows in the base fact will not be aggregated unless the included filter is true.  This is potentially costly in performance and global filters should always be preferred to dynamic filters, if possible.
+
+The `AS` parameter name, if supported, will create an alias for the dynamic metric which will be used in the schema of the response as well as in the having clause of the request.
+
+Example:
+
+```
+?metrics=users(filter,dim1|id-in['a','b'],AS=DimOneInAOrB)&having=DimOneInAOrB-gt[50]
+```
+
+This would create a variation of the standard `users` aggregation which only counted users from records where the dimension `dim1` had values "a" or "b", and would display result rows only which for which this metric had at least 50 users.
```

Comment:
Will the response show metric as `DimOneInAOrB` instead of `users`?

Reply:
yes